### ISSUE TITLE
[iOS] Causes folly::toJson Exception when used in conjunction with react-native-ivs

### ISSUE THREAD

**Issue by kyle-ssg** [software-mansion/react-native-reanimated#1776]
## Description

I'm using react-native-aws-ivs-player-view to play video, for some reason as soon as I install this library it causes a knock on effect where the app crashes due to folly trying to parse some json that has NaN.

If I uninstall react-native-reanimated from the project, the app runs fine without this crash. 

I fixed the issue causing this for the particular library [here](https://github.com/perrystreetsoftware/react-native-aws-ivs-player-view/pull/8) . I do appreciate that this is a knock-on effect due perhaps bad Native iOS code, however some information as to why this package could throw up these crashes would be good if it isn't fixable.

```
libc++abi.dylib: terminating with uncaught exception of type folly::json::parse_error: folly::toJson: JSON object value was a NaN or INF
```

### Screenshots
![image](https://user-images.githubusercontent.com/8608314/109545232-8181a980-7ac0-11eb-9e17-83f05c3d06dc.png)

### Expected behavior
react-native-reanimated does not cause knock on folly exceptions for other libraries, or there is explanation why this could happen.
### Actual behavior
Installing react-native-reanimated caused the mentioned exception

## Snack or minimal code example

Install react-native-reanimated alongside https://github.com/perrystreetsoftware/react-native-aws-ivs-player-view/tree/main/example

## Package versions

    "react": "16.13.1",
    "react-native": "0.63.4",
    "react-native-reanimated": "2.0.0-rc.3"

**Comment by mrbrentkelly** [comment]
I'm getting the exact same issue in a custom built Video Player native module.

```
libc++abi.dylib: terminating with uncaught exception of type folly::json::parse_error: folly::toJson: JSON object value was a NaN or INF
terminating with uncaught exception of type folly::json::parse_error: folly::toJson: JSON object value was a NaN or INF
```

**Comment by Deodes** [comment]
The same thing with [react-native-video](https://github.com/react-native-video/react-native-video) and reanimated..

**Comment by tomekzaw** [comment]
Hello! Can you please verify if this is still an issue on Reanimated 2.3.1?

**Comment by tomekzaw** [comment]
Here's the only occurrence of `folly::toJson` in our codebase:

https://github.com/software-mansion/react-native-reanimated/blob/fc837335e017e53473123d4aaedf3502cc0d901c/ios/native/NativeProxy.mm#L249

This call throws `folly::json::parse_error` exception because `NaN` and `Inf` values cannot be serialized to JSON.

In Reanimated, we need to override `RCTEventDispatcher` with `REAEventDispatcher` in order to intercept all events and handle some of them synchronously. Here's the overriden implementation of `sendEvent` method:

https://github.com/software-mansion/react-native-reanimated/blob/2c489378fc0b4e1a92ed87102fa3f79f36710e27/ios/REAEventDispatcher.m#L8-L12

As one can see, we don't have any mechanism of selecting whether the event should be processed by Reanimated module or handled over to JS. I imagine the improved implementation to be:

```objc
if (/* event can be handled by ReanimatedModule */) {
  [[[self bridge] moduleForName:@"ReanimatedModule"] eventDispatcherWillDispatchEvent:event]; 
} else {
   [super sendEvent:event]; 
}
```

As a workaround, we can wrap `folly::toJson` call within `try/catch` block. This way we can prevent the crashes for third-party events while keeping the solution minimal and avoid breaking things up.

**Comment by tomekzaw** [comment]
Fixed with #2896
